src/filter-log.py

```diff
@@ -0,0 +1,434 @@
+
+
+################################################################################
+# PARSE ARGS AND MAIN
+################################################################################
+def regexp(astring):
+    try:
+        re.compile(astring)
+        return astring
+    except re.error:
+        raise argparse.ArgumentTypeError("Argument must be a valid regular expression: " + astring)
+
+
+def get_args() -> argparse.Namespace:
+    """
+    Configure the argument parsing system, and run it, to obtain the arguments given on the commandline.
+    :return: The parsed arguments.
+    """
+    parser = argparse.ArgumentParser(formatter_class=argparse.RawDescriptionHelpFormatter)
+    parser.add_argument("-d", "--debug", action="store_true",
+                        help="Output extra debug information")
+    parser.add_argument("-v", "--verbose", action="store_true",
+                        help="Output extra verbose information. Implies --debug")
+#    parser.add_argument("-g", "--groups", default="",
+#                        type=regexp,
+#                        help="Use groups for some components. [<none>]")
+    parser.add_argument("-o", "--output",
+                        type=str,
+                        default="log",
+                        help="Stem of output file [log]")
+    parser.add_argument("-f", "--folder",
+                        type=str,
+                        default=".",
+                        help="Folder to write logfiles to [.]")
+    parser.description = "Filter kafka json lines, extract key entries from message field, and write to 'rotated' files."
+    parser.epilog = """
+Examples:
+    TBD:
+    """ + sys.argv[0] + """ 
+
+"""
+    args = parser.parse_args()
+    return args
+
+class MyDumper(yaml.Dumper):
+    def increase_indent(self, flow=False, indentless=False):
+        return super(MyDumper, self).increase_indent(flow, False)
+
+
+def main():
+    start_time = datetime.datetime.now()
+    try:
+        global script_name
+        args = get_args()
+
+        global do_debug
+        global do_trace
+        global do_dryrun
+        do_debug = args.debug
+        # do_trace = args.trace
+        do_verbose = args.verbose
+        if do_verbose:
+            do_debug = True
+
+        debug("cli options: debug:" + str(do_debug)
+              + ", verbose: " + str(do_verbose)
+              + ", output: " + str(args.output)
+              + ", folder: " + str(args.folder)
+              )
+
+        message_rx = re.compile("\s+")
+        output_file = None
+        for line in fileinput.input('-'):
+
+            try:
+                blob = json.loads(line)
+            except ValueError as e:
+                # Check if this is an empty line
+                line = line.strip()
+                if not line:
+                    continue
+
+                # It wasn't empty. Now it is stripped - try again.
+                try:
+                    blob = json.loads(line)
```

Comment:
Every other line is blank - how about moving
"strip / continue" right after "for", to avoid trying to parse them